File: plsql-squid/src/test/java/br/com/felipezorzo/sonar/plsql/statements/CommitStatementTest.java
package br.com.felipezorzo.sonar.plsql.statements;

import static org.sonar.sslr.tests.Assertions.assertThat;

import org.junit.Before;
import org.junit.Test;

import br.com.felipezorzo.sonar.plsql.api.PlSqlGrammar;
import br.com.felipezorzo.sonar.plsql.api.RuleTest;

public class CommitStatementTest extends RuleTest {

    @Before
    public void init() {
        setRootRule(PlSqlGrammar.COMMIT_STATEMENT);
    }
    
    @Test
    public void matchesSimpleCommit() {
        assertThat(p).matches("commit;");
    }
    
    @Test
    public void matchesCommitWork() {
        assertThat(p).matches("commit work;");
    }
    
    @Test
    public void matchesCommitForce() {
        assertThat(p).matches("commit force 'test';");
    }
    
    @Test
    public void matchesCommitForceWithScn() {
        assertThat(p).matches("commit force 'test',1;");
    }
    
    @Test
    public void matchesCommitWithComment() {
        assertThat(p).matches("commit comment 'test';");
    }
    
    @Test
    public void matchesCommitWrite() {
        assertThat(p).matches("commit write;");
    }
    
    @Test
    public void matchesCommitWriteImmediate() {
        assertThat(p).matches("commit write immediate;");
    }
    
    @Test
    public void matchesCommitWriteBatch() {
        assertThat(p).matches("commit write batch;");
    }
    
    @Test
    public void matchesCommitWriteWait() {
        assertThat(p).matches("commit write wait;");
    }
    
    @Test
    public void matchesCommitWriteNoWait() {
        assertThat(p).matches("commit write nowait;");
    }
    
    @Test
    public void matchesLongCommitStatement() {
        assertThat(p).matches("commit work comment 'teste' write immediate wait;");
    }
    
    @Test
    public void matchesLabeledCommit() {
        assertThat(p).matches("<<foo>> commit;");
    }

}

File: zpa-core/src/test/kotlin/com/felipebz/zpa/api/statements/CommitStatementTest.kt
/**
 * Z PL/SQL Analyzer
 * Copyright (C) 2015-2025 Felipe Zorzo
 * mailto:felipe AT felipezorzo DOT com DOT br
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
package com.felipebz.zpa.api.statements

import com.felipebz.flr.tests.Assertions.assertThat
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import com.felipebz.zpa.api.PlSqlGrammar
import com.felipebz.zpa.api.RuleTest

class CommitStatementTest : RuleTest() {

    @BeforeEach
    fun init() {
        setRootRule(PlSqlGrammar.COMMIT_STATEMENT)
    }

    @Test
    fun matchesSimpleCommit() {
        assertThat(p).matches("commit;")
    }

    @Test
    fun matchesCommitWork() {
        assertThat(p).matches("commit work;")
    }

    @Test
    fun matchesCommitForce() {
        assertThat(p).matches("commit force 'test';")
    }

    @Test
    fun matchesCommitForceWithScn() {
        assertThat(p).matches("commit force 'test',1;")
    }

    @Test
    fun matchesCommitWithComment() {
        assertThat(p).matches("commit comment 'test';")
    }

    @Test
    fun matchesCommitWrite() {
        assertThat(p).matches("commit write;")
    }

    @Test
    fun matchesCommitWriteImmediate() {
        assertThat(p).matches("commit write immediate;")
    }

    @Test
    fun matchesCommitWriteBatch() {
        assertThat(p).matches("commit write batch;")
    }

    @Test
    fun matchesCommitWriteWait() {
        assertThat(p).matches("commit write wait;")
    }

    @Test
    fun matchesCommitWriteNoWait() {
        assertThat(p).matches("commit write nowait;")
    }

    @Test
    fun matchesLongCommitStatement() {
        assertThat(p).matches("commit work comment 'teste' write immediate wait;")
    }

    @Test
    fun matchesLabeledCommit() {
        assertThat(p).matches("<<foo>> commit;")
    }

}
